girder/models/item.py

```diff
@@ -501,7 +501,11 @@ def fileList(self, doc, user=None, path='', includeMetadata=False,
         metadataFile = 'girder-item-metadata.json'
 
         fileModel = File()
-        for file in self.childFiles(item=doc):
+        # Eagerly evaluate this list, as the MongoDB cursor can time out on long requests
+        # Don't use a "filter" projection here, since returning the full file document is promised
```

Comment:
Note this.

Reply:
No worries, these are the leaf nodes, so they won't be on the stack for long.